### PR TITLE
Added channel description to API and UI

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Part of #263 

#### What's this PR do?
Adds the channel description to the API and renders it as markdown on the channel page.

#### How should this be manually tested?
- Go to /api/v0/channels/:channel_name/ and verify you see the description field returned.
- Update the channel description with some markdown like (not all these markdown examples will work, I was lazy and copy/pasted an excessive example):
```python
from django.contrib.auth.models import User
from channels.api import Api
user = User.objects.get(username="mitodl")
api = Api(user)
api.update_channel("channel_name", description="""
An h1 header
============

Paragraphs are separated by a blank line.

2nd paragraph. *Italic*, **bold**, and `monospace`. Itemized lists
look like:

  * this one
  * that one
  * the other one

Note that --- not considering the asterisk --- the actual text
content starts at 4-columns in.

> Block quotes are
> written like so.
>
> They can span multiple paragraphs,
> if you like.

Use 3 dashes for an em-dash. Use 2 dashes for ranges (ex., "it's all
in chapters 12--14"). Three dots ... will be converted to an ellipsis.
Unicode is supported. ☺

An h2 header
------------

Here's a numbered list:

 1. first item
 2. second item
 3. third item

### An h3 header ###

Now a nested list:

 1. First, get these ingredients:

      * carrots
      * celery
      * lentils
2. hiiiii

Notice again how text always lines up on 4-space indents (including
that last line which continues item 3 above).

Here's a link to [a website](http://foo.bar), to a [local
doc](local-doc.html), and to a [section heading in the current
doc](#an-h2-header). 

A horizontal rule follows.

***

Here's a definition list:

apples
  : Good for making applesauce.
oranges
  : Citrus!
tomatoes
  : There's no "e" in tomatoe.

Again, text is indented 4 spaces. (Put a blank line between each
term/definition pair to spread things out more.)

Here's a "line block":

| Line one
|   Line too
| Line tree

""")
````
- Verify this content and/or the default value is correctly rendered in the sidebar.
- Ensure no adverse affects on other pages in the app (e.g. pages can load and layout is unaffected by changes to support right sidebar).

#### Screenshots

localhost_8063_channel_odl

### CODE REVIEW

**Comment by aliceriot** [comment]
functionality looks good, reading code now.

File: static/js/components/ChannelSidebar_test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+describe("ChannelSidebar", () => {
+  const renderSidebar = channel => shallow(<ChannelSidebar channel={channel} />)
+
+  it("should render sidebar", () => {
+    const channel = makeChannel()
+    const wrapper = renderSidebar(channel)
+    const [description] = wrapper.find(ReactMarkdown)
+    assert.equal(description.props.source, channel.description)
+  })
```
Comment (aliceriot) [inline static/js/components/ChannelSidebar_test.js]: a basic test that "There is no description..." shows up when `channel.description` is empty might be worthwhile here

File: static/js/containers/HomePage.js [hunk 2]
```diff
@@ -94,6 +94,8 @@ const mapStateToProps = state => {
   }
 }
 
-export default R.compose(connect(mapStateToProps), withNavSidebar, withLoading)(
-  HomePage
-)
+export default R.compose(
+  connect(mapStateToProps),
+  withOnlyNavSidebar,
```
Comment (aliceriot) [inline static/js/containers/HomePage.js]: suggestion you can take or leave: to me it makes a little more semantic sense to have `withNavSidebar` and `withNavAndChannelSidebars` or similar, those names seem more accurate to me

File: static/js/hoc/withNavSidebar.js [hunk 3]
```diff
@@ -1,30 +1,8 @@
...
-            <WrappedComponent {...this.props} />
-          </div>
-        </div>
-      )
-    }
-  }
-  return WithNavSidebar
-}
-
-export default withNavSidebar
```
Comment (aliceriot) [inline static/js/hoc/withNavSidebar.js]: I might suggest we simplify this a little bit and just declare two HOCs (`withNavSidebar` and `withNavAndChannelSidebars`) that just take a `WrappedComponent` and render stuff directly - as is there's kind of a lot of indirection going on that I don't quite see the need for. Figuring out what `withNavSidebar = withSidebars(NavSidebar)` means, for instance, requires quite a bit of poking around. My thinking is that right now we only have two variations (nav and nav + channel description) and I don't think we're likely to have more anytime soon.

File: channels/factories.py [hunk 4]
```diff
@@ -152,27 +157,47 @@ def get_instances(self, factory_type):
...
+            strategy (str): either a create or build strategy for the factory
+            kwargs (dict): the kwargs to pass to the factory
+
+        Returns:
+            object: the created object
+        """
+        if strategy == STRATEGY_CREATE:
+            return factory_cls.create(**kwargs)
+        elif strategy == STRATEGY_BUILD:
+            return factory_cls.build(**kwargs)
```
Comment (aliceriot) [inline channels/factories.py]: what's the difference between `.create` and `.build` here?

Comment (rhysyngsun) [reply channels/factories.py]: `factoryboy` explains the difference here: http://factoryboy.readthedocs.io/en/latest/reference.html#strategies

TLDR: `create` means persist, `build` means don't

**Comment by rhysyngsun** [comment]
@aliceriot feedback addressed

File: channels/factories.py (hunk 4 above)
Comment (aliceriot) [reply channels/factories.py]: Ok I figured that was what was going on, sounds good.

**Comment by aliceriot** [comment]
Looking good now, the only thing I just noticed is that the spacing between the navigation sidebar and the list of posts is now slightly different on the homepage than it is on the subreddit page.

homepage:

homepagespacing

subredditspacing

these images are different sizes so it's maybe a little hard to tell but on the subreddit there's more space between them.

**Comment by rhysyngsun** [comment]
@aliceriot I fixed that with a fixed padding around main-content. Note: the layout *may* appear to jump if your homepage scrolls but your channel page doesn't or vice versa.
